src/tests/Microsoft.PowerFx.Core.Tests/CheckResultTests.cs

```diff
@@ -258,6 +258,53 @@ public void BindingCheckReturnType()
             Assert.Equal("Error 0-3: The type of this expression does not match the expected type 'Text'. Found type 'Number'.", error.ToString());
         }
 
+        [Fact]
```

Comment:
[Theory]
        [InlineData("\"str\"", false, true)] // trivial, String == String 
        [InlineData("1", false, false)] // Number != String        
        [InlineData("1", true, true)] // Number can coerce to String
        [InlineData("{x:1}", true, false)] // Record can't be coerce to String
        public void SetExpectedReturnValueString(string inputExpr, bool allowCoerce, bool succeed)
        {
            var check = new CheckResult(new Engine())
                .SetText(inputExpr)
                .SetBindingInfo()
                .SetExpectedReturnValue(FormulaType.String, allowCoerce);

            var errors = check.ApplyErrors();

            if (succeed)
            {
                Assert.True(check.IsSuccess);
            }
            else
            {
                Assert.False(check.IsSuccess);

                Assert.Single(errors);
                var error = errors.First();
                Assert.Equal("Error 0-15: The type of this expression does not match the expected type 'Text'. Found type 'Color'.", error.ToString());
            }
        }

Reply:
add these tests